### DISCUSSION TITLE
View errors thrown in API hooks (graphql)

### DISCUSSION THREAD

**Discussion by mooori** [directus/directus#7622] [Archive]
Are those errors logged somewhere respectively is it possible to find out about errors that were thrown?

Let's say the following hook is registered (taken from the [docs](https://docs.directus.io/guides/api-hooks/#before)):

```javascript
module.exports = function registerHook({ exceptions }) {
	const { InvalidPayloadException } = exceptions;

	return {
		'items.create.before': async function (input) {
			if (LOGIC_TO_CANCEL_EVENT) {
				throw new InvalidPayloadException(WHAT_IS_WRONG);
			}

			return input;
		},
	};
};
```

If I execute a graphql mutation `create_articles_item` which causes `InvalidPayloadException` to be thrown, it seems like I cannot find out about the exception being thrown.

The graphql response is simply `create_articles_item: null` with http status `200`.

The directus log is (`LOG_LEVEL="debug"`):
`request completed POST 200 /graphql`

In the activity log the request doesn't show up (no item was created).

Did I miss something?

**Edit** It's possible to do `console.log` or some other form of login from within the hook. Though, I was wondering if there's a built-in way to see the error given the handler function above.

**Comment by rijkvanzanten** [comment]
GraphQL always responds with a 200 status, so that's expected. It should have an `errors` array though for any errors thrown within the query execution.

**Reply by mooori** [reply]
I've been checking the response body, it's `{data: {create_articles_item: null}}`, no `errors` or other fields.

**Reply by rijkvanzanten** [reply]
Mind opening a new issue about that? Errors you throw in a custom before hook should propagate out to the graphql output 👍🏻

**Reply by mooori** [reply]
Done: #7651. Thank you.
